### PR TITLE
parser: Support window function identifiers

### PR DESCRIPTION
**Pull request by wangyisong1996**
Add reserved and nonreserved identifiers related to window function.
The `ROW` identifier has become reserved.
See https://dev.mysql.com/doc/refman/8.0/en/window-functions.html and https://dev.mysql.com/doc/refman/8.0/en/keywords.html for details.

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -148,6 +151,9 @@ import (
 	key			"KEY"
 	keys			"KEYS"
 	kill			"KILL"
+	lag				"LAG"
```
Comment (hanfei1991) [inline parser/parser.y]: an extra `\t` char

File: parser/parser.y [hunk 2]
```diff
@@ -193,6 +204,9 @@ import (
 	revoke			"REVOKE"
 	right			"RIGHT"
 	rlike			"RLIKE"
+	row				"ROW"
```
Comment (hanfei1991) [inline parser/parser.y]: ditto.

**Comment by hanfei1991** [comment]
LGTM

**Comment by zz-jason** [comment]
@winkyao @coocood PTAL

**Comment by shenli** [comment]
LGTM

**Comment by wangyisong1996** [comment]
@shenli @winkyao @coocood PTAL, thanks!

**Comment by wangyisong1996** [comment]
Please ... DO NOT merge at this moment. There're still some keywords forgotten.

**Comment by shenli** [comment]
Is this PR ready to merge?

**Comment by hanfei1991** [comment]
@shenli yes

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
MyBatis use `groups` as a table name. But it is an reserved keyword in MySQL 8.0 which is used for window function.
We will fix MyBatis latter.

**Pull request merged**